docs/strategies/python/poetry.md

```diff
@@ -0,0 +1,516 @@
+# Poetry
+
+[Poetry](https://python-poetry.org/) is a tool for dependency management and packaging in Python.
+
+## Project Discovery
+
+Find all files named `pyproject.toml` and `poetry.lock`. Pyproject must use poetry for the build system.
+
+## Analysis
+
+We parse `pyproject.toml` -- to find direct dependencies and thier [version constraints](https://python-poetry.org/docs/dependency-specification/):
+
+- `[tool.poetry.dependencies]` - production dependencies
+- `[tool.poetry.dev-dependencies]` - development dependencies
+
+To infer resolved version of direct and sub dependencies, we parse `poetry.lock` file. In this file we find,
+
+- `[package.dependencies]` - package's dependencies 
+- `package.category` - package's environment (dev, test, etc.)
+- `package.name` - name of the package
+- `package.version` - resolved version of the package
+
+For projects, where `poetry.lock` file is not discovered, we fallback to reporting only direct dependencies parsed from `pyproject.toml`. 
+
+### Limitations
+
+* For poetry project, build system's `build-backend` must be set to `poetry.core.masonry.api` in `pyproject.toml` 
```

Comment:
If the user is using poetry project - this is done by default when initializing the project. It would be extremely rare to use poetry - and not have this configured. I've added poetry docs referencing build-backend parameter in latest commit. I've also cleared the ambiguity around what happens if the `buid-backend` is not set to `poetry.core.masonry.api` in the project discovery.